sdks/java/io/hadoop-format/src/main/java/org/apache/beam/sdk/io/hadoop/format/HadoopFormatIO.java

```diff
@@ -437,6 +445,12 @@
           .build();
     }
 
+    /** Transforms the keys read from the source using the given key translation function. */
+    public Read<K, V> withKeyTranslation(SimpleFunction<?, K> function, Coder<K> coder) {
```

Comment:
Since we don't need the type information anymore, we should use SerializableFunction instead of SimpleFunction.

Reply:
I did try to pass in `SerializableFunction` but inner code of HadoopFormat does require input and output types for the function for validation, e.g. that input format value class does match translation function input class. I choose to stick with `SimpleFunction` but just allow to pass in the specific coder.